src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -264,9 +264,12 @@ void DatasetManager::HandleSet(Coap::Header &aHeader, Message &aMessage, const I
             aMessage.Read(offset + sizeof(Tlv), sizeof(timestamp), &timestamp);
         }
 
-        // verify the request does not include fields that affect connectivity
-        if (tlvType == Tlv::kChannel || tlvType == Tlv::kMeshLocalPrefix ||
-            tlvType == Tlv::kPanId || tlvType == Tlv::kNetworkMasterKey)
+        // verify the request only includes dataset tlvs that do not affect connectivity
+        if (tlvType != Tlv::kActiveTimestamp && tlvType != Tlv::kPendingTimestamp &&
+            tlvType != Tlv::kChannelMask && tlvType != Tlv::kExtendedPanId &&
+            tlvType != Tlv::kNetworkName && tlvType != Tlv::kPSKc &&
+            tlvType != Tlv::kSecurityPolicy && tlvType != Tlv::kDelayTimer &&
+            tlvType != Tlv::kCommissionerSessionId)
```

Comment:
I believe the spec allows TLVs not defined in the spec to be included (to allow for forwarding compatibility in the case that a new TLV is defined in the future).